<h1><b>Assignment: Online Shop</b></h1>

<b><h4>Orders Table</h4></b>

In [6]:
import pandas as pd

orders_df = pd.read_csv("orders.csv")

def display_info(orders_df):
    print("\nFirst 30 rows of the dataset:")
    display(orders_df.head(30))  
    print(f"\nTotal Rows: {orders_df.shape[0]}")
    print(f"Total Columns: {orders_df.shape[1]}")

display_info(orders_df)


First 30 rows of the dataset:


order_id  order_date  customer_id  total_price
0       7324  2023-11-27         2160       955.86
1       8938  2024-08-31         8497       984.91
2       4238  2023-12-05         6295       649.18
3      10944  2024-02-22         2936        54.83
4      11075  2024-08-14         2994       320.24
5      13842  2024-05-20         5953       555.05
6       1126  2023-11-17         4825       308.21
7       8310  2024-08-09         7873       957.72
8       6084  2024-08-23         5627       934.52
9       3161  2024-11-01         2086       488.16
10        57  2024-05-26          975       388.08
11     11179  2024-02-06         7024        24.32
12      1860  2024-09-30         4276        42.33
13      9492  2024-09-27         4374        43.54
14      2853  2024-03-29         1888        44.45
15        51  2024-05-16         3177       735.84
16     11997  2024-04-25         5410       620.96
17      6792  2024-05-21         7971       812.89
18      9279  2024-04-26          995       241.51
19      5292  2024-08-03         1259       123.95
20      8357  2023-11-05         4733       206.41
21     14041  2024-05-10         7404       459.13
22      4657  2024-07-25         9938       423.73
23       100  2024-01-11         3089       111.23
24      1802  2024-08-31         9808       677.87
25      7789  2023-12-03         8797       465.42
26      3325  2024-09-04         4478       475.69
27      2843  2023-12-26         6169       432.37
28     11500  2024-09-15         7978       675.62
29      6897  2024-07-12         6618       516.94


Total Rows: 15000
Total Columns: 4


In [8]:
orders_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 15000 entries, 0 to 14999
Data columns (total 4 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   order_id     15000 non-null  int64  
 1   order_date   15000 non-null  object 
 2   customer_id  15000 non-null  int64  
 3   total_price  15000 non-null  float64
dtypes: float64(1), int64(2), object(1)
memory usage: 468.9+ KB


**Datatype of order_date is not relevant**

In [14]:
orders_df.describe()

order_id  customer_id   total_price
count  15000.000000  15000.00000  15000.000000
mean    7500.500000   4982.93820   2833.046987
std     4330.271354   2886.06956   2275.527394
min        1.000000      1.00000     20.460000
25%     3750.750000   2477.75000    932.520000
50%     7500.500000   4974.50000   2207.065000
75%    11250.250000   7476.00000   4249.185000
max    15000.000000  10000.00000  10082.720000

**Range beetween min and max is huge, so there is a chance of occurence of outliers**

In [20]:
orders_df.isnull()

order_id  order_date  customer_id  total_price
0         False       False        False        False
1         False       False        False        False
2         False       False        False        False
3         False       False        False        False
4         False       False        False        False
...         ...         ...          ...          ...
14995     False       False        False        False
14996     False       False        False        False
14997     False       False        False        False
14998     False       False        False        False
14999     False       False        False        False

[15000 rows x 4 columns]

In [22]:
orders_df.isnull().sum()

order_id       0
order_date     0
customer_id    0
total_price    0
dtype: int64

In [24]:
orders_df['order_date'].value_counts()

order_date
2024-07-31    58
2024-07-20    56
2024-10-11    56
2024-01-03    56
2024-10-12    55
              ..
2024-05-28    28
2023-12-29    23
2023-11-15    21
2023-11-05    17
2024-11-04    15
Name: count, Length: 366, dtype: int64

<h1><b>Major Assignment: Steam Review&Games Dataset</b></h1>

<h2><b>About Dataset</b></h2>

The dataset provides a collection of game reviews from the Steam platform, making it suitable for natural language processing (NLP) and machine learning projects. The columns included are:

<br><b>id</b>: Unique identifier for the review. 
<br><b>app_id</b>: Game identifier on Steam.
<br><b>content</b>: Review text.
<br><b>author_id</b>: Identifier for the review's author.
<br><b>is_positive</b>: Label indicating whether the review is positive (1) or negative (0).

In [50]:
import pandas as pd

df = pd.read_csv("output.csv")

def display_info(df):
    print("\nFirst 20 rows of the dataset:")
    display(df.head(20))  
    print(f"\nTotal Rows: {df.shape[0]}")
    print(f"Total Columns: {df.shape[1]}")


# Display information about the dataset
display_info(df)


First 20 rows of the dataset:


id  app_id                                            content  \
0   181331361     100               At least its a counter strike -1/100   
1   180872601     100  Uh... So far my playthrough has not been great...   
2   177836246     100                          Better mechanics than cs2   
3   177287444     100              buggy mess and NOT fun to play at all   
4   176678990     100  Whoever came up with this, is gonna fucking ge...   
5   176118004     100  Every enemy and their grandmother have aimbot ...   
6   172896379     100                                                  l   
7   171821329     100  Counter Strike: Condition Zero Deleted Scenes(...   
8   171639855     100  [REVIEW IS FOR DELETED SCENES] Game is pretty ...   
9   167078851     100                                   too many enemies   
10  166120260     100                                                  .   
11  165309201     100                                              weird   
12  163565928     100  classsic doom style of shitty level design, un...   
13  162309591     100         I cant find the key in the first level!!!1   
14  161797645     100                              It's just not for me.   
15  161782841     100  𝗩𝗮𝗹𝘃𝗲'𝘀 "𝗗𝗲𝘀𝘁𝗶𝗻𝗲𝗱 𝘁𝗼 𝗯𝗲 𝗮 𝗳𝗮𝗶𝗹𝘂𝗿𝗲" 𝗰𝗵𝗶𝗹𝗱\r\n\r...   
16  161525162     100  cs campaing sound cool but the game just spawn...   
17  159882018     100  there are some missions that are good but hone...   
18  159514632     100           Interesting idea but terrible execution.   
19  156138045     100                                      why just why?   

            author_id is_positive  
0   76561199556485100    Negative  
1   76561199230620391    Negative  
2   76561198417690647    Negative  
3   76561199077268730    Negative  
4   76561199104544266    Negative  
5   76561198296823531    Negative  
6   76561199182060159    Negative  
7   76561199032983213    Negative  
8   76561198866570651    Negative  
9   76561199554114911    Negative  
10  76561198951396265    Negative  
11  76561198092967794    Negative  
12  76561198171195751    Negative  
13  76561199224381226    Negative  
14  76561198967539132    Negative  
15  76561198966155773    Negative  
16  76561198962753813    Negative  
17  76561197975930687    Negative  
18  76561199251818153    Negative  
19  76561198028295408    Negative


Total Rows: 201151
Total Columns: 5


In [42]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 201151 entries, 0 to 201150
Data columns (total 5 columns):
 #   Column       Non-Null Count   Dtype 
---  ------       --------------   ----- 
 0   id           201151 non-null  int64 
 1   app_id       201151 non-null  int64 
 2   content      200723 non-null  object
 3   author_id    201151 non-null  int64 
 4   is_positive  201151 non-null  object
dtypes: int64(3), object(2)
memory usage: 7.7+ MB


**Non-null indicates that all of them have values.**
<br>**Object which means a string.**
<br>**All the ids are int which is a number.**
<br>**No problem with datatypes**

In [44]:
df.describe()

id         app_id     author_id
count  2.011510e+05  201151.000000  2.011510e+05
mean   1.224624e+08     452.899722  7.656120e+16
std    5.896154e+07     322.187389  5.423023e+08
min    2.090000e+02      10.000000  7.656120e+16
25%    7.292665e+07     240.000000  7.656120e+16
50%    1.479831e+08     440.000000  7.656120e+16
75%    1.690541e+08     620.000000  7.656120e+16
max    1.822350e+08    1930.000000  7.656120e+16

**Excluded all object files and computed math operations using IDs**
<br>**Doesn't make any sense in this case)**

In [61]:
df.isnull().sum()

id               0
app_id           0
content        428
author_id        0
is_positive      0
dtype: int64

**Here we see that content attribute has 428 null cells which means that they are empty**

In [67]:
# Count occurrences of 'negative' and 'positive' in the 'is_positive' column
negative_count = df['is_positive'].str.contains('Negative', case=False, na=False).sum()
positive_count = df['is_positive'].str.contains('Positive', case=False, na=False).sum()

# Display results
print(f"Number of 'negative': {negative_count}")
print(f"Number of 'positive': {positive_count}")


Number of 'negative': 98491
Number of 'positive': 102660


**We see that number of positive reviews exeeds negative ones**

In [72]:
# Group by 'app_id' and calculate counts for 'negative' and 'positive'
result = df.groupby('app_id')['is_positive'].apply(
    lambda group: pd.Series({
        'negative_count': group.str.contains('negative', case=False, na=False).sum(),
        'positive_count': group.str.contains('positive', case=False, na=False).sum()
    })
).reset_index()

# Display the result
print(result)

    app_id         level_1  is_positive
0       10  negative_count         1555
1       10  positive_count         9900
2       20  negative_count          669
3       20  positive_count         4665
4       30  negative_count          304
..     ...             ...          ...
95    1840  positive_count           91
96    1900  negative_count          206
97    1900  positive_count           67
98    1930  negative_count          605
99    1930  positive_count           83

[100 rows x 3 columns]


**Here we see number of negative and positive reviews for each distinct app_id**